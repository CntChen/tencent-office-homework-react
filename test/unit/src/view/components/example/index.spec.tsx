import React from 'react';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';
import Example from 'src/view/components/example';

/**
 * @author cntchen
 * @priority P0
 * @casetype unit
 */
describe('Example Component Test', () => {
    afterEach(cleanup);

    it('传递属性渲染', () => {
        const tilteProp = 'Hello Test Title';

        render(
            <Example
                {...{
                    title: tilteProp,
                }}
            />,
        );

        const titleWrap = screen.getAllByText(new RegExp(tilteProp, 'i'));
        // 可以通过 screen.debug 获得 DOM 格式化输出
        // screen.debug(titleWrap);
        expect(titleWrap).not.toBeUndefined();
    });

    it('点击按钮逻辑', () => {
        render(
            <Example
                {...{
                    title: 'test',
                }}
            />,
        );

        const countSpan = screen.getByText('0', {
            selector: 'span',
        });
        const clickButton = screen.getByText(/click me/i, {
            selector: 'button',
        });

        expect(countSpan.innerHTML).toEqual('0');
        expect(clickButton).toBeEnabled();
        fireEvent.click(clickButton);
        expect(countSpan.innerHTML).toEqual('1');
    });

    it('Snapshot', () => {
        const { container } = render(
            <Example
                {...{
                    title: 'test',
                }}
            />,
        );

        expect(container.innerHTML).toMatchSnapshot();
    });
});
