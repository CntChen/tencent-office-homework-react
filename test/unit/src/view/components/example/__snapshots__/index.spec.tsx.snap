// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Component Test Snapshot 1`] = `
<div class="hook-example">
  <div>
    Hook Example
  </div>
  <p class="count-number">
    In test page, You clicked
    <span>
      0
    </span>
    times
  </p>
  <button>
    Click me
  </button>
</div>
`;
